consensus/src/dag/mod.rs

```diff
@@ -0,0 +1,641 @@
+//! Converts units from the network into ones that are in the Dag, in the correct order.
+use std::collections::HashMap;
+
+use crate::{
+    alerts::{Alert, ForkingNotification},
+    units::{
+        SignedUnit, UncheckedSignedUnit, Unit, UnitStore, Validator as UnitValidator, WrappedUnit,
+    },
+    Data, Hasher, MultiKeychain,
+};
+use log::{debug, trace, warn};
+
+mod reconstruction;
+mod validation;
+
+pub use reconstruction::{ReconstructedUnit, Request};
+use reconstruction::{Reconstruction, ReconstructionResult};
+pub use validation::ValidatorStatus as DagStatus;
+use validation::{Error as ValidationError, Validator};
+
+const LOG_TARGET: &str = "AlephBFT-dag";
+
+/// The result of sending some information to the Dag.
+pub struct DagResult<H: Hasher, D: Data, MK: MultiKeychain> {
+    units: Vec<ReconstructedUnit<SignedUnit<H, D, MK>>>,
+    requests: Vec<Request<H>>,
+    alerts: Vec<Alert<H, D, MK::Signature>>,
+}
+
+impl<H: Hasher, D: Data, MK: MultiKeychain> DagResult<H, D, MK> {
+    fn empty() -> Self {
+        DagResult {
+            units: Vec::new(),
+            requests: Vec::new(),
+            alerts: Vec::new(),
+        }
+    }
+
+    fn alert(alert: Alert<H, D, MK::Signature>) -> Self {
```

Comment:
```suggestion
    fn from_alert(alert: Alert<H, D, MK::Signature>) -> Self {
```
?

Reply:
Eh, it's supposed to be read as "dag result that is an alert" rather than "...made from an alert", and this is also how `ReconstructionResult` constructors are named.

Reply:
fair enough